src/main/java/org/opensearch/knn/index/codec/transfer/VectorTransfer.java

```diff
@@ -5,50 +5,43 @@
 
 package org.opensearch.knn.index.codec.transfer;
 
-import lombok.Data;
-import org.apache.lucene.util.BytesRef;
-import org.opensearch.knn.index.codec.util.SerializationMode;
+import java.io.Closeable;
+import java.io.IOException;
 
 /**
- * Abstract class to transfer vector value from Java to native memory
+ * An interface to transfer vectors from one memory location to another
+ * Class is Closeable to be able to release memory once done
  */
-@Data
-public abstract class VectorTransfer {
-    protected final long vectorsStreamingMemoryLimit;
-    protected long totalLiveDocs;
-    protected long vectorsPerTransfer;
-    protected long vectorAddress;
-    protected int dimension;
-
-    public VectorTransfer(final long vectorsStreamingMemoryLimit) {
-        this.vectorsStreamingMemoryLimit = vectorsStreamingMemoryLimit;
-        this.vectorsPerTransfer = Integer.MIN_VALUE;
-    }
+public interface VectorTransfer extends Closeable {
 
     /**
-     * Initialize the transfer
-     *
-     * @param totalLiveDocs total number of vectors to be transferred
+     * Transfer a single vector from one location to another
+     * @throws IOException
      */
-    abstract public void init(final long totalLiveDocs);
+    void transfer() throws IOException;
 
     /**
-     * Transfer a single vector
-     *
-     * @param bytesRef a vector in bytes format
+     * Transfer a batch of vectors from one location to another
+     * The batch size here is intended to be constant for multiple transfers so should be encapsulated in the
+     * implementation. A new batch size should require another instance
+     * @throws IOException
      */
-    abstract public void transfer(final BytesRef bytesRef);
+    void transferBatch() throws IOException;
```

Comment:
transfer will transfer one vector at a time

Reply:
why we need to transfer 1 vector at a time?

Reply:
> Its intended to transfer one vector at a time, this will be useful when we are doing iterative build on addValue

It should be handled when we are building that functionality. Because at that time we can focus on that functionality in more details around what is needed. Like what if we want to reuse batch=1 and transferLimit=1 to send 1 vector at time, rather than this. I may not be thinking optimal here, but I do think inserting 1 vector at a time will have more things to look at hence its better to not add this functionality. 

Since this is my opinion. I am happy to know more on this.

Reply:
I can remove it for now